Added Support for Fahrenheit and Kelvin units with an example

ashutoshshrm529 opened the pull request:
I have added support for fahrenheit and kelvin since they are commonly used and should be in-built in the library

ashutoshshrm529 committed:
Added Support for Fahrenheit and Kelvin units

ashutoshshrm529 committed:
Respositioned the example to fit arduino standard.

per1234 commented on examples/ReadSensorOtherUnits/ReadSensorOtherUnits.ino:
```diff
@@ -0,0 +1,47 @@
+  Serial.print(THERM.readTemperature(FAHRENHEIT));
+  Serial.println(" °F");
+
+  Serial.print("Temperature in Kelvin");
```
```suggestion
  Serial.print("Temperature in Fahrenheit ");
  Serial.print(THERM.readTemperature(FAHRENHEIT));
  Serial.println(" °F");

  Serial.print("Temperature in Kelvin ");
```
The temperature value was printed right up against the end of the words.

per1234 commented on examples/ReadSensorOtherUnits/ReadSensorOtherUnits.ino:
```diff
@@ -0,0 +1,47 @@
+/*
+  MKR THERM Shield - Read Sensors
+
+  This example reads the temperatures (in fahrenheit and kelvin) measured by the thermocouple
```
```suggestion
  This example reads the temperatures (in Fahrenheit and kelvins) measured by the thermocouple
```

per1234 commented on examples/ReadSensorOtherUnits/ReadSensorOtherUnits.ino:
```diff
@@ -0,0 +1,47 @@
+
+  Serial.print("Temperature in Kelvin");
+  Serial.print(THERM.readTemperature(KELVIN));
+  Serial.println(" °F");
```
```suggestion
  Serial.println(" K");
```
Use correct unit symbol.

per1234 commented on src/MKRTHERM.cpp:
```diff
@@ -96,7 +96,18 @@ float THERMClass::readTemperature()
+  }
+  else if(units=CELSIUS)
+  {
+      return celsius;
```
```suggestion
  if (units == FAHRENHEIT) {
    return (celsius * 9.0 / 5.0) + 32.0;
  } else if(units == KELVIN) {
    return (celsius + 273.15);
  } else if(units == CELSIUS) {
    return celsius;
```
- Fix bug caused by the use of assignment operator instead of comparison operator.
- Make code formatting consistent with the rest of the file.

per1234 reviewed (dismissed):
Please always thoroughly test your code before submitting a pull request.

---
DONE ~~You have introduced a compiler warning:~~
```
MKRTHERM.cpp:111:1: warning: control reaches end of non-void function [-Wreturn-type]
```
My instinct is to add `return NAN;` to the end of the function, but I took a look at the Arduino_HTS221 library and it just defaults to returning the Celsius value:
https://github.com/arduino-libraries/Arduino_HTS221/blob/0f309b2633f89bd44a1779d43b39448c5665527b/src/HTS.cpp#L89-L93
I'll leave it up to you to decide how to fix the warning.

---
DONE ~~Please add the `FAHRENHEIT`, `CELCIUS`, and `KELVIN` keywords to keywords.txt, as was done in the Arduino_HTS221 library:~~
https://github.com/arduino-libraries/Arduino_HTS221/blob/0f309b2633f89bd44a1779d43b39448c5665527b/keywords.txt#L25-L26
Be sure to use a true tab field separator, not spaces.
https://arduino.github.io/arduino-cli/library-specification/#keywordstxt-format

ashutoshshrm529 committed:
Update examples/ReadSensorOtherUnits/ReadSensorOtherUnits.ino

Co-Authored-By: per1234 <accounts@perglass.com>

Update src/MKRTHERM.cpp

Co-Authored-By: per1234 <accounts@perglass.com>

Update examples/ReadSensorOtherUnits/ReadSensorOtherUnits.ino

Co-Authored-By: per1234 <accounts@perglass.com>

Update examples/ReadSensorOtherUnits/ReadSensorOtherUnits.ino

Co-Authored-By: per1234 <accounts@perglass.com>

Made the function default to celcius as suggested.

ashutoshshrm529 commented:
@per1234 ,

> Please always thoroughly test your code before submitting a pull request.
> 
> You have introduced a compiler warning:
> 
> ```
> MKRTHERM.cpp:111:1: warning: control reaches end of non-void function [-Wreturn-type]
> ```
> 
> My instinct is to add `return NAN;` to the end of the function, but I took a look at the Arduino_HTS221 library and it just defaults to returning the Celsius value:
> https://github.com/arduino-libraries/Arduino_HTS221/blob/0f309b2633f89bd44a1779d43b39448c5665527b/src/HTS.cpp#L89-L93
> I'll leave it up to you to decide how to fix the warning.

I have made the function default to the celsius setting similar to the HTS library.

> Please add the `FAHRENHEIT`, `CELCIUS`, and `KELVIN` keywords to keywords.txt, as was done in the Arduino_HTS221 library:
> https://github.com/arduino-libraries/Arduino_HTS221/blob/0f309b2633f89bd44a1779d43b39448c5665527b/keywords.txt#L25-L26
> Be sure to use a true tab field separator, not spaces.
> https://arduino.github.io/arduino-cli/library-specification/#keywordstxt-format

I have added the keywords. I apologise for not adding them before since I was not aware.
Please recheck the tab setting (I added a single tab but I am not sure about my editor's autoindent.)




also, @per1234 , I am applying for GSoC this year and have written a draft proposal for the same, since you seem to be a very active member of this community, I would be very grateful if you could  please review my proposal and suggest changes

ashutoshshrm529 committed:
Had Forgotten to push Keywords.txt

aentinger reviewed (approved):
LGTM :+1:
